api/src/services/relations.ts

```diff
@@ -229,7 +229,7 @@ export class RelationsService {
 								this.schema.collections[existingRelation.related_collection!].primary
 							}`
 						)
-						.onDelete(relation.schema?.on_delete || 'NO ACTION');
+						.onDelete(relation.schema?.on_delete || undefined!);
```

Comment:
`undefined!` is a bit of a weird pattern in TS 🤔 

It's basically saying "undefined, but it's not undefined! 🤫"

Reply:
It certainly is, but we get a type error any other way. It's logically similar to the `.inTable(constraint.one_collection!)` in the migration file, but because of the null propagation we can't just do `relation.schema?.on_delete!`.

Personally I dislike this workaround, but I'd seen similar behavior elsewhere in the codebase. I can reset the branch to 10c4ac8 if you prefer that solution?

Reply:
> but I'd seen similar behavior elsewhere in the codebase.

Yeah.. as a matter of fact, I just had to do a very similar hack-around for another knex type issue..

https://github.com/directus/directus/blob/c3b74a86f10cceecfceeb1607d5512877410bfa2/api/src/utils/apply-query.ts#L306-L309

Reply:
> I can reset the branch to 10c4ac8 if you prefer that solution?

Do we conclusively know what Knex defaults to in case of `undefined`? Is it different per database?

Reply:
I can't say with total confidence. The default is to just ignore it if onDelete is falsy: https://github.com/knex/knex/blob/1744c8c2655a8362260ba987d706ec3473681a78/lib/schema/tablecompiler.js#L96

I don't see any of the schemas overriding the `foreign` function, so it seems to be safe.

Logically it makes sense to me that `undefined` would be valid in this scenario. The function takes the `onDelete` value and stores it as metadata to be used in the compile step. The default of `onDelete` would already be `undefined`, so setting it again to `undefined` should change nothing? 🤔

Reply:
Ohhh I see, it just creates the constraint without any on delete rules. That makes sense 👍🏻

Reply:
Yeah, exactly. And you'd hope omitting the on delete rule would be handled by all databases... 😬

Anywho, does this code look alright or your you prefer 10c4ac8?

Reply:
Should we try omitting the whole `onDelete` call completely instead of passing `undefined`? That might be slightly more explicit 🤔 

```js
if (relation.schema?.on_delete) {
  table.onDelete(relation.schema.on_delete);
}
```

I'm also fine to just do it like this tho, no need to bikeshed on it too much. Up to you!

Reply:
I'll revert to commit 10c4ac8 then since that just omits the onDelete call?